tests/scripts/generate_test_code.py

```diff
@@ -800,44 +798,45 @@ def parse_test_data(data_f):
                                   "%s" % (data_f.name, data_f.line_no, name))
 
 
-def gen_dep_check(dep_id, dep):
+def gen_dependencies_element(dep):
     """
-    Generate code for checking dependency with the associated
-    identifier.
+    Generate an entry in the dependency array.
 
-    :param dep_id: Dependency identifier
     :param dep: Dependency macro
-    :return: Dependency check code
+    :return: Dependency array element initializer
     """
-    if dep_id < 0:
-        raise GeneratorInputError("Dependency Id should be a positive "
-                                  "integer.")
-    _not, dep = ('!', dep[1:]) if dep[0] == '!' else ('', dep)
-    if not dep:
+    _not, core_dep = ('!', dep[1:]) if dep[0] == '!' else ('', dep)
+    if not core_dep:
         raise GeneratorInputError("Dependency should not be an empty string.")
 
-    dependency = re.match(CONDITION_REGEX, dep, re.I)
+    dependency = re.match(CONDITION_REGEX, core_dep, re.I)
     if not dependency:
         raise GeneratorInputError('Invalid dependency %s' % dep)
 
     _defined = '' if dependency.group(2) else 'defined'
     _cond = dependency.group(2) if dependency.group(2) else ''
     _value = dependency.group(3) if dependency.group(3) else ''
 
-    dep_check = '''
-        case {id}:
-            {{
-#if {_not}{_defined}({macro}{_cond}{_value})
-                ret = DEPENDENCY_SUPPORTED;
+    pp_expr = '{_not}{_defined}({macro}{_cond}{_value}'.format(
+        _not=_not, _defined=_defined, macro=dependency.group(1),
+        _cond=_cond, _value=_value)
+    dep_check = '''\
+#if {pp_expr})
```

Comment:
> Also, I'm thinking it might be clearer to separate the "defined" case from the "condition" (comparison operator) case. Perhaps even use different regexes?

Maybe but that's preexisting and out of scope.

> why are some variable names starting with an underscore?

`not` is a keyword, so at a guess `_not` is to avoid that keyword. But the [Python convention](https://peps.python.org/pep-0008/#descriptive-naming-styles) for that is `not_`. I don't know about the other, maybe for consistency with `not`, but I don't find that a good reason.